.github/workflows/staging-deployment.yml

```
@@ -0,0 +1,179 @@
+name: Artemis Staging Deployment
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Branch to deploy'
+        required: true
+
+concurrency: staging
+
+env:
+  build_workflow_name: build.yml
+
+jobs:
+  check-build-status:
+    runs-on: ubuntu-latest
+    outputs:
+      build_workflow_run_id: ${{ steps.set_build_workflow_id.outputs.workflow_id }} 
+    steps:
+      - name: Get latest build workflow run
+        id: get_workflow_run
+        uses: octokit/request-action@v2.x
+        with:
+          route: GET /repos/${{ github.repository }}/actions/workflows/${{ env.build_workflow_name }}/runs?branch=${{ github.event.inputs.branch }}&per_page=1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set build workflow ID
+        id: set_build_workflow_id
+        run: |
+          WORKFLOW_ID=$(echo '${{ steps.get_workflow_run.outputs.data }}' | jq -r '.workflow_runs[0].id')
+          echo "workflow_id=$WORKFLOW_ID" >> $GITHUB_OUTPUT
+  
+      - name: Check for war artifact
+        id: verify_artifact
+        uses: octokit/request-action@v2.x
+        with:
+          route: GET /repos/${{ github.repository }}/actions/runs/${{ steps.set_build_workflow_id.outputs.workflow_id }}/artifacts?name=Artemis.war
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  
+      - name: Verify artifact exists
+        id: check_result
+        run: |
+          TOTAL_COUNT=$(echo '${{ steps.verify_artifact.outputs.data }}' | jq -r '.total_count')
+            
+          if [ "$TOTAL_COUNT" -gt 0 ]; then
+            echo "Found Artemis.war artifact in latest build"
+          else
+            echo "::error::No Artemis.war artifact found in latest build!"
+            exit 1
+          fi
+
+  deploy:
+    needs: check-build-status
+    runs-on: github-l-4
+    environment: 
+        name: artemis-staging-localci.artemis.cit.tum.de
+        url: ${{ vars.DEPLOYMENT_URL }}
+    env:
+      DEPLOYMENT_HOSTS_PRIMARY: ${{ vars.DEPLOYMENT_HOSTS_PRIMARY}}
+      DEPLOYMENT_HOSTS_SECONDARY: "artemis-staging-localci-node-2.artemis.cit.tum.de artemis-staging-localci-node-3.artemis.cit.tum.de artemis-staging-localci-node-4.artemis.cit.tum.de artemis-staging-localci-node-5.artemis.cit.tum.de"
```

Please use the environment variable here:
```suggestion
      DEPLOYMENT_HOSTS_SECONDARY: ${{ vars.DEPLOYMENT_HOSTS_SECONDARY }}
```